src/viam/sdk/CMakeLists.txt

```diff
@@ -0,0 +1,204 @@
+# Declare the viamcpp library, and attach sources,
+# headers for installation
+add_library(viamsdk)
+
+# TODO: This is analogized from
+# https://gitlab.kitware.com/cmake/cmake/-/tree/master/Help/guide/importing-exporting/MathFunctionsComponents
+# but it doesn't really seem to do anything. Everything works fine if
+# it is removed, and naming the alias something other than `viamcpp`
+# doesn't make it available under that alias anyway, as one might
+# hope. What is it supposed to do?
+add_library(viam-cpp-sdk::viamsdk ALIAS viamsdk)
+
+# This bit of magic is required so that we get the public build-time
+# include path of viamapi in *ahead* of the path anchored to `src` by
+# `BASE_DIRS` below, because otherwise we will pull in the static
+# versions of the proto files even if we are generating dynamically.
+target_include_directories(viamsdk
+  PUBLIC
+    "$<BUILD_INTERFACE:$<TARGET_PROPERTY:viam-cpp-sdk::viamapi,INTERFACE_INCLUDE_DIRECTORIES>>"
+)
```

Comment:
Subtle and fragile.

This cost me a bunch of time. When we made everything have a common include root, that meant that when we added the build-time include path for `cppsdk` it would end up injecting the static proto generated headers in front of the dynamic ones. This bit ensures that at viamsdk build time that we place the path to the dynamic protos first.